### PR TITLE
8267338: [JVMCI] revive JVMCI API removed by JDK-8243287

### PR DESCRIPTION
**Pull request by dougxc**
This PR revives ResolvedJavaType.getHostClass to preserve JVMCI compatibility. The revived method just returns `null`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8267338](https://bugs.openjdk.java.net/browse/JDK-8267338): [JVMCI] revive JVMCI API removed by JDK-8243287

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**) ⚠️ Review applies to 5bffb99ae302fd5a8808c9404948791787680ed1
 * [Harold Seigel](https://openjdk.java.net/census#hseigel) (@hseigel - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4099/head:pull/4099` \
`$ git checkout pull/4099`

Update a local copy of the PR: \
`$ git checkout pull/4099` \
`$ git pull https://git.openjdk.java.net/jdk pull/4099/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4099`

View PR using the GUI difftool: \
`$ git pr show -t 4099`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4099.diff">https://git.openjdk.java.net/jdk/pull/4099.diff</a>

</details>

### CODE REVIEW

File: src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.meta/src/jdk/vm/ci/meta/ResolvedJavaType.java [hunk 1]
```diff
@@ -136,6 +136,14 @@ default boolean declaresDefaultMethods() {
      */
     boolean isAssignableFrom(ResolvedJavaType other);
 
+    /**
+     * Returns {@code null} since support for VM anonymous class was removed by JDK-8243287.
+     * This method is preserved for JVMCI backwards compatibility.
+     */
+    default ResolvedJavaType getHostClass() {
```
Comment (christianwimmer) [inline src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.meta/src/jdk/vm/ci/meta/ResolvedJavaType.java:143]: Mark the method as `deprecated`?

**Review by mlchung — APPROVED** [review]
I missed this in my review.  Thanks for adding it back.

**Comment by vnkozlov** [comment]
It seems the fix is incomplete - next test is failing: compiler/jvmci/jdk.vm.ci.runtime.test/src/jdk/vm/ci/runtime/test/TestResolvedJavaType.java

**Review by hseigel — APPROVED** [review]
LGTM
Thanks, Harold

**Pull request closed without merging**